### PR TITLE
Fix issue with competing SDK proxy configuration

### PR DESCRIPTION
**Pull request by breedloj**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
A recent [change](https://github.com/aws/aws-sdk-java-v2/commit/3748964d2793d732eae6dfead9691327fdd14569) in the AWS SDK introduced the ability to provide proxy configuration via environment variables. Since the Toolkit primarily relies on proxy configuration provided by the host IDE configuration this introduces a potential conflict when configuration exists in both places, leading to an SDK exception (see #4279).

This change disables the SDK from picking up proxy configuration from environment variables meaning that configuration will only be inherited from IDE settings. Additionally the SDK version is bumped to the latest. 

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changes/next-release/bugfix-c8090208-8e8a-4a41-bcdd-02579f74d40e.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "Fix issue with competing SDK proxy configuration"
```
Comment (rli) [inline .changes/next-release/bugfix-c8090208-8e8a-4a41-bcdd-02579f74d40e.json:3]:
```suggestion
  "description" : "Fix issue with competing SDK proxy configuration (#4279)"
```

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/AwsSdkClient.kt [hunk 2]
```diff
@@ -24,7 +24,12 @@ class AwsSdkClient : SdkClientProvider, Disposable {
     private val sdkHttpClient: SdkHttpClient by lazy {
         LOG.info { "Create new Apache client" }
         val httpClientBuilder = ApacheHttpClient.builder()
-            .proxyConfiguration(ProxyConfiguration.builder().useSystemPropertyValues(false).build())
+            .proxyConfiguration(
+                ProxyConfiguration.builder()
+                    .useSystemPropertyValues(false)
+                    .useEnvironmentVariableValues(false)
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/AwsSdkClient.kt:30]: test case would be nice too

Comment (breedloj) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/AwsSdkClient.kt:30]: Added tests that assert the client functions as expected when system property or env var proxy config is passed.

File: plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/AwsSdkClientTest.kt [hunk 3]
```diff
@@ -63,6 +68,42 @@ class AwsSdkClientTest {
...
+            System.clearProperty("http.proxyHost")
+            System.clearProperty("http.proxyPort")
+        }
+
+        assertThat(response.httpResponse().isSuccessful).isTrue()
+    }
+
+    @Test
+    fun environmentVariableProxyConfigurationIgnored() {
+        environmentVariableHelper.set("HTTP_PROXY", "http://foo.com:8888")
```
Comment (rli) [inline plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/AwsSdkClientTest.kt:92]: maybe should be HTTPS

File: plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/AwsSdkClientTest.kt [hunk 4]
```diff
@@ -63,6 +68,42 @@ class AwsSdkClientTest {
...
+
+        val request = mockSdkRequest("https://localhost:" + wireMock.httpsPort())
+
+        val response = try {
+            AwsSdkClient().sharedSdkClient().prepareRequest(
+                HttpExecuteRequest.builder().request(request).build()
+            ).call()
+        } finally {
+            System.clearProperty("http.proxyHost")
+            System.clearProperty("http.proxyPort")
```
[the remark this answers is not in the document]
Comment (breedloj) [reply plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/core/AwsSdkClientTest.kt:84]: Good catch & point - makes a lot more sense why we're using these. 👍

**Pull request merged**